### DISCUSSION TITLE
2023/06/20/leetcode%E5%88%B7%E9%A2%98%E8%AE%B0%E5%BD%9515-LRU%20%E7%BC%93%E5%AD%98/

### DISCUSSION THREAD

**Discussion by giscus** [Zhuanjier/blog#20] [Announcements]
# 2023/06/20/leetcode%E5%88%B7%E9%A2%98%E8%AE%B0%E5%BD%9515-LRU%20%E7%BC%93%E5%AD%98/

¶LRU 缓存  原题传送门  给定两个大小分别为m和n的正序（从小到大）数组nums1和nums2。请你找出并返回这两个正序数组的中位数。 请你设计并实现一个满足LRU (最近最少使用) 缓存约束的数据结构。 实现LRUCache类：  LRUCache(int capacity)以正整数作为容量capacity初始化 LRU 缓存 `int get(int key) 如果关键字 key 存在于

https://blog.zhuanjie.ltd/2023/06/20/leetcode%E5%88%B7%E9%A2%98%E8%AE%B0%E5%BD%9515-LRU%20%E7%BC%93%E5%AD%98/

**Comment by RandallAnjie** [comment]
在C++中，`unordered_map`是一个关联容器，它提供了快速查找的能力。你可以用它来存储键值对，并通过键来快速查找到相应的值。通过哈希表查找键的复杂度为O(1)，在哈希表中插入元素的复杂度也都是O(1)。
`unordered_map::find(key)`是一个成员函数，它会试图在`unordered_map`中找到一个与给定键匹配的元素。如果找到了，它会返回一个指向这个元素的迭代器。如果没有找到，那么它会返回一个特殊的迭代器，这个迭代器等于`unordered_map::end()`，`unordered_map::end()`函数返回一个指向`unordered_map`末尾（实际上是容器“尾后”位置，即最后一个元素之后的位置）的迭代器。
因此，`cache.find(key) == cache.end()`的意思就是尝试在`unordered_map`中查找键`key`，如果没找到（即返回了指向`unordered_map`末尾的迭代器），则条件为真，否则条件为假。
也就是说，`cache.find(key) == cache.end()`的意思就是“在缓存中查找键`key`，如果找不到则返回真，如果找到则返回假”。
